activesupport/test/dependencies_test.rb

```diff
@@ -240,6 +240,46 @@ class AnotherConstant::ReloadError
     end
   end
 
+  # Regression see https://github.com/rails/rails/issues/32359
+  def test_explicit_unloadable_constants_can_be_unloaded_and_reloaded
+    with_loading "autoloading_fixtures/explicit_unloadable" do
+      # autoload this constant:
+      assert UnloadableExample
+
+      # Assert that AS::D's internal state matches Ruby's state:
+      assert defined?(UnloadableExample)
+      assert_equal ["UnloadableExample"], ActiveSupport::Dependencies.autoloaded_constants
+      assert_equal 1, ActiveSupport::Dependencies.explicitly_unloadable_constants.size
+      assert_equal 1, ActiveSupport::Dependencies.loaded.size
+      # Now cleanly unload the constant
+      ActiveSupport::Dependencies.remove_unloadable_constants!
+
+      # Since the constant was properly unloaded, it can be reloaded:
+      assert UnloadableExample
+    end
+  end
+
+  # Regression see https://github.com/rails/rails/issues/32359
+  def test_explicit_unloadable_constants_can_are_cleanly_unloaded
+    with_loading "autoloading_fixtures/explicit_unloadable" do
+      # autoload this constant:
+      assert UnloadableExample
+
+      # Assert that AS::D's internal state matches Ruby's state:
+      assert defined?(UnloadableExample)
+      assert_equal ["UnloadableExample"], ActiveSupport::Dependencies.explicitly_unloadable_constants
+      assert_equal ["UnloadableExample"], ActiveSupport::Dependencies.autoloaded_constants
+      assert_equal 1, ActiveSupport::Dependencies.loaded.size
+
+      # Unload it, and do a reality check that Ruby's state and AS::D's state match
+      ActiveSupport::Dependencies.remove_unloadable_constants!
+      refute defined?(UnloadableExample), "The constant was actually removed"
```

Comment:
Prefer `assert_not` to `refute` (http://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#follow-the-coding-conventions)

Reply:
:+1: Definitely, thanks for the link!